lib/parsers/moov_parser.rb

```diff
@@ -115,5 +117,35 @@ def matches_moov_definition?(io)
     maybe_atom_size >= minimum_ftyp_atom_size && maybe_ftyp_atom_signature == 'ftyp'
   end
 
+  # Sample information is found in the 'time-to-sample' stts atom.
+  # The media atom mdhd is needed too in order to get the movie timescale
+  def parse_sample_atom(decoder, atom_tree)
+    video_trak_atom = decoder.find_video_trak_atom(atom_tree)
+
+    stts = if video_trak_atom
+      decoder.find_first_atom_by_path([video_trak_atom], 'trak', 'mdia', 'minf', 'stbl', 'stts')
+    else
+      decoder.find_first_atom_by_path(atom_tree, 'moov', 'trak', 'mdia', 'minf', 'stbl', 'stts')
+    end
+
+    mdhd = if video_trak_atom
+      decoder.find_first_atom_by_path([video_trak_atom], 'trak', 'mdia', 'mdhd')
+    else
+      decoder.find_first_atom_by_path(atom_tree, 'moov', 'trak', 'mdia', 'mdhd')
+    end
+
+    if stts && mdhd
+      timescale = mdhd.atom_fields[:tscale]
+      sample_duration = stts.field_value(:entries).first[:sample_duration]
+      if timescale.nil? || timescale == 0 || sample_duration.nil? || sample_duration == 0
+        nil
+      else
+        timescale.to_f / sample_duration
```

Comment:
How does float division paired with the above truncation work for some odd frame_rates? I suspect we'd get some pretty weird values sometimes. Curious if you tried running some videos through other tools and check what they return for the frame rate.

Reply:
I tried to look for weird videos and I saw in our fixture folder (within the spec folder) has already some videos. One of these has a weird frame rate of 14.98 and the tool calculates that properly.. if this is what you meant!

Reply:
If you've checked that, then I guess we're fine. It's just odd using `truncate` in the context of floats, was expecting something more in the lines of `round`.